rfd/0094-discover-metrics.md

```diff
@@ -0,0 +1,195 @@
+    string type = 1;
+    // Optional protocol where applicable, e.g. postgres, mysql, sqlserver.
+    string protocol = 2;
+}
+
+// Contains fields that track a particular step outcome, for example connection
+// test failed or succeeded, or user aborted the step.
+message Status {
+    // Indicates the step outcome. For example, "success" means user proceeded
+    // to the next step, "error" if something failed (e.g. connection test),
+    // "aborted" if user exited the wizard.
+    string status = 1;
+    // Contains error details. We have to be careful to not include any
+    // identifyable infomation like server addresses here.
+    string error = 2 ;
+}
+
+// Emitted when user selected resource type to add and proceeded to the next
+// step.
+message DiscoverResourceSelectionEvent {
+    DiscoverMetadata discover_metadata = 1;
+    ResourceMetadata resource_metadata = 2;
+    Status status = 3;
+}
+
+// Emitted on the "Configure Resource" screen when user has connected an SSH
+// or Kubernetes agent and proceeded to the next step.
+//
+// For Database Access this step has sub-steps so will have individual events
+// defined below, however this event will be emitted as well at the end of
+// the last sub-step to be able to track it on a more high-level.
+message DiscoverConfigureResourceEvent {
+    DiscoverMetadata discover_metadata = 1;
+    ResourceMetadata resource_metadata = 2;
+    Status status = 3;
+}
+
+// Emitted when a user registered a database resource and proceeded to the next
+// step.
+message DiscoverRegisterDatabase {
+    DiscoverMetadata discover_metadata = 1;
+    ResourceMetadata resource_metadata = 2;
+    Status status = 3;
+}
+
+// Emitted when a user deployed a database agent and proceeded to the next step.
+message DiscoverDeployDatabaseAgent {
+    DiscoverMetadata discover_metadata = 1;
+    ResourceMetadata resource_metadata = 2;
+    Status status = 3;
+}
+
+// Emitted when a user configured mutual TLS for self-hosted database and
+// proceeded to the next step.
+message DiscoverConfigureMTLS {
+    DiscoverMetadata discover_metadata = 1;
+    ResourceMetadata resource_metadata = 2;
+    Status status = 3;
+}
+
+// Emitted when a user configured IAM for RDS database and proceeded to the
+// next step.
+message DiscoverConfigureIAM {
+    DiscoverMetadata discover_metadata = 1;
+    ResourceMetadata resource_metadata = 2;
+    Status status = 3;
+}
+
+// Emitted on "Setup Access" screen when user has updated their principals
+// and proceeded to the next step.
+message DiscoverSetUpAccessEvent {
+    DiscoverMetadata discover_metadata = 1;
+    ResourceMetadata resource_metadata = 2;
+    Status status = 3;
+}
+
+// Emitted on the "Test Connection" screen when user clicked tested connection
+// to their resource.
+message DiscoverTestConnectionEvent {
+    DiscoverMetadata discover_metadata = 1;
+    ResourceMetadata resource_metadata = 2;
+    Status status = 3;
```

Comment:
Should we add the failing traces in case of failure?
I think they would be really useful: was it a RBAC check, connectivity issue, host/db refused the login user

Reply:
Yes, the idea was that this information will make it in the "error" field on the "status".